projects/dotcms-field-elements/src/components/dot-textfield/dot-texfield.e2e.ts

```diff
@@ -1,140 +1,330 @@
-                name: 'fullName',
-                status: {
-                    dotPristine: true,
-                    dotTouched: true,
-                    dotValid: true
-                }
-            });
-        });
-
-        it('should send status and value change', async () => {
-            await input.press('a');
-            await page.waitForChanges();
-            expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
-                name: 'fullName',
-                status: {
-                    dotPristine: false,
-                    dotTouched: true,
-                    dotValid: true
-                }
-            });
-            expect(spyValueChange).toHaveReceivedEventDetail({
-                name: 'fullName',
-                value: 'Johna'
-            });
-        });
-
-        it('should emit status and value on Reset', async () => {
-            await element.callMethod('reset');
-            expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
-                name: 'fullName',
-                status: {
-                    dotPristine: true,
-                    dotTouched: false,
-                    dotValid: false
-                }
-            });
-            expect(spyValueChange).toHaveReceivedEventDetail({ name: 'fullName', value: '' });
+        describe('disabled', () => {
+            it('should render disabled attribute', async () => {
+                element.setProperty('disabled', 'true');
+                await page.waitForChanges();
+                expect(input.getAttribute('disabled')).toBeDefined();
+            });
+
+            it('should not render disabled attribute', async () => {
+                element.setProperty('disabled', 'false');
+                await page.waitForChanges();
+                expect(input.getAttribute('disabled')).toBeNull();
+            });
+
+            it('should render disabled attribute with invalid value', async () => {
+                element.setProperty('disabled', { test: 'test' });
+                await page.waitForChanges();
+                expect(input.getAttribute('disabled')).toBeDefined();
+            });
+        });
+
+        describe('type', () => {
+            it('should set value to text on default correctly', async () => {
+                await page.waitForChanges();
+                expect(input.getAttribute('type')).toBe('text');
+            });
+
+            it('should set value correctly', async () => {
+                element.setProperty('type', 'email');
+                await page.waitForChanges();
+                expect(input.getAttribute('type')).toBe('email');
+            });
+
+            it('should render and not break when is a unexpected value', async () => {
+                element.setProperty('type', { test: true });
+                await page.waitForChanges();
+                expect(input.getAttribute('type')).toBe('[object Object]');
```

Comment:
When we set an invalid type to a native `input` it sets the default one

<img width="397" alt="Screen Shot 2019-05-23 at 3 12 55 PM" src="https://user-images.githubusercontent.com/751424/58287028-4986f780-7d6d-11e9-9943-7cce5067b429.png">

Reply:
that is the natural behavior of the input, it sets to `text` no matter if initially the value was 'email'. Think we don't have to test that scenario.

Reply:
No what I'm saying is that the input type should behave the same as the native one